encrypted-media/scripts/playback-persistent-license.js

```diff
@@ -0,0 +1,104 @@
+function runTest(config, testname) {
+
+    var testname = config.keysystem + ', successful playback, persistent-license, '
+                                    + /video\/([^;]*)/.exec( config.videoType )[ 1 ]
+                                    + ', set src before setMediaKeys';
+
+    var configuration = {   initDataTypes: [ config.initDataType ],
+                            audioCapabilities: [ { contentType: config.audioType } ],
+                            videoCapabilities: [ { contentType: config.videoType } ],
+                            sessionTypes: [ 'persistent-license' ] };
+
+
+    async_test( function( test ) {
+
+        var _video = config.video,
+            _mediaKeys,
+            _mediaKeySession,
+            _mediaSource,
+            _releaseSequence = false;
+
+        function onFailure(error) {
+            forceTestFailureFromPromise(test, error);
+        }
+
+        function onMessage(event) {
+            assert_equals( event.target, _mediaKeySession );
+            assert_true( event instanceof window.MediaKeyMessageEvent );
+            assert_equals( event.type, 'message');
+
+            if ( !_releaseSequence )
+            {
+                assert_in_array( event.messageType, [ 'license-request', 'individualization-request' ] );
+            }
+            else
+            {
+                assert_equals( event.messageType, 'license-release' );
+            }
+
+            config.messagehandler( config.keysystem, event.messageType, event.message )
+            .then( function( response ) {
+                _mediaKeySession.update( response )
+                .catch(onFailure);
+            });
+        }
+
+        function onEncrypted(event) {
+            assert_equals(event.target, _video);
+            assert_true(event instanceof window.MediaEncryptedEvent);
+            assert_equals(event.type, 'encrypted');
+
+            waitForEventAndRunStep('message', _mediaKeySession, onMessage, test);
+            _mediaKeySession.generateRequest(   config.initData ? config.initDataType : event.initDataType,
```

Comment:
Just to be sure that we always supply both the config variable or both the event variables and never one of each.